packages/nimble-components/src/spinner/styles.ts

```diff
@@ -18,6 +18,7 @@ export const styles = css`
     }
 
     div.container {
+        animation-play-state: inherit;
```

Comment:
Nervous about this since it's pulling from arbitrary up in the dom hierarchy.

Maybe instead an overrideable private property

```css
:host {
    --ni-private-spinner-animation-play-state: running;
}
div.container {
    animation-play-state: var(--ni-private-spinner-animation-play-state);
}
```
And the story can overrite the private property. (maybe double check the private css property name convention, didn't actually double check)

Reply:
Can you help me understand this concern? The behavior I'm seeing is that `animation-play-state` is not inherited by default, so a client must set it on this div's parent (i.e. `nimble-spinner`) directly for it to affect this div. What do you mean by "arbitrarily up in the dom hierarchy"?

I'm not opposed to the private property approach, just want to learn why.

Reply:
Ah I was wrong, think my mental model was specifying a property as `inherit` causes it to go up to the first actual property definition but it just goes up to the computed value of the parent element: https://developer.mozilla.org/en-US/docs/Web/CSS/Inheritance

I guess specifying the default on host would make me feel better in my (incorrect) mental model but is technically not necessary:

```css
:host {
    animation-play-state: running;
}
```

So then the question is do we actually want to expose the animation-play-state as part of the public api? i.e. if users set a value for that css property we want to respond vs [marking it's configuration forbidden](https://github.com/ni/nimble/issues/598).

If yes I think there is only one possible animated part on this control so what you did looks good. I guess the question is then do we want to make that CSS API functional anywhere else we have animations?

Also we may also want to consider the use-case. I think it's unlikely that a user would just want to disable animations on a single control and it would be handy to configure that across multiple controls easily. If that is possibly true maybe we want a design token that can be shared across many controls and is configured by the theme provider. These aren't mutually exclusive either, we just would use the default to come from the token so it could be overridden locally (which is actually how we should be using tokens like [controlHeight, see option 2](https://github.com/ni/nimble/issues/610)).

```css
:host {
     animation-play-state: ${animationPlayStateToken};
}
```

If we don't want to expose this stuff on the public API then my suggestion shows how to keep it internal.

Reply:
Thanks for thinking about it. The rationale of needing to think about our public CSS API more holistically resonates with me. It doesn't really make sense to respect `animation-play-state` for just this control so I went with a private custom property.

Reply:
Also regarding
>maybe double check the private css property name convention,
it turns out we didn't document a convention beyond "prefix with --ni-private". I added a note about prefixing with the element name. I considered linking to the [token naming guidelines](https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#token-naming) but they felt less applicable for private variables to me since they're explicitly not part of the public API.